homeassistant/components/light/child_light.py

```diff
@@ -0,0 +1,195 @@
+    EVENT_PLATFORM_DISCOVERED, {
+        ATTR_SERVICE: DISCOVER_CHILD_LIGHTS,
+        ATTR_DISCOVERED: data})
+
+
+This example will monitor the value of 'brightness' in the parent component's
+'state.attributes' dictionary.  The state card will ned be visible on the UI
+and available to automations etc.
+
+The 'state_attributes' is also optional but if a dictionary is passed in the
+items will be added to the child sensors state attributes.
+
+Whenever 'turn_on' or 'turn_off' are called the service specified in
+'parent_service' will be called.
+
+The 'extra_data' field will be passed to the service in the service data
+dictionary and can be very useful for establishing context in the service
+callback.
+
+The 'parent_action' field is also passed back to the service when called and
+can be used to distinguish between different types of service requests if
+necessary.
+
+The 'light_type' field can be either 'dimmer' or 'switch' depending on the
+type of light.
+
+NOTES:
+
+- At the moment this can only be created via the 'discovery' mechanism.  So do
+not add it into your config file
+
+- You need to add 'light' or 'child_light' as a dependency for your component
+otherwise there is no guarantee this component will be loaded when the
+discovery event fires.
+
+"""
+
+from homeassistant.components.switch.child_switch import ChildSwitch
+from homeassistant.helpers import event
+from homeassistant.components.light import (
+    Light, ATTR_BRIGHTNESS, ATTR_XY_COLOR)
+from homeassistant.const import (
+    STATE_ON, STATE_OFF, ATTR_ENTITY_ID)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the lights. """
+    dev = []
+    child_light = ChildLight(config, discovery_info)
+    dev.append(child_light)
+
+    add_devices(dev)
+
+    event.track_state_change(
+        hass,
+        discovery_info.get('parent_entity_id'),
+        child_light.track_state)
+
+
+class ChildLight(ChildSwitch, Light):
+    """ This class is a light that can be created by another component
+        it monitors the state of the parent device and calls the specified
+        service when toggling is initiated """
+
+    def __init__(self, config, discovery_info=None):
+        super().__init__(config, discovery_info)
+        self._xy = 0
+        self._brightness = discovery_info.get('initial_brightness')
+        self._light_type = discovery_info.get('light_type', 'switch')
+        if self._light_type == 'dimmer':
+            if self._brightness == 0:
+                self._state = STATE_OFF
+            else:
+                self._state = STATE_ON
+
+    @property
+    def brightness(self):
+        """ Brightness of this light between 0..255. """
+        return self._brightness
+
+    @property
+    def color_xy(self):
+        """ XY color value. """
+        return self._xy
+
+    def turn_on(self, **kwargs):
+        """ Turn the entity on. """
+        self._state = STATE_ON
```

Comment:
The idea of the child platforms is that they expose a attribute of another state as a entity. This means that it should not contain any logic for optimistic updates as this means that the child entity is out of sync with its parent. If an optimistic update has to be done, it should be done by the parent and pushed to the childs via a state change.